Webpack css-loader failing at @import '~node_module/path/to/css.css'

<!-- Please don't delete this template or we'll close your issue -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->
<!-- Also consider trying the webpack@beta version, maybe it's already fixed. -->

**Do you want to request a *feature* or report a *bug*?**
bug
<!-- Please ask questions on StackOverflow or the webpack Gitter (https://gitter.im/webpack/webpack). -->
<!-- Issues which contain questions or support requests will be closed. -->

**What is the current behavior?**
In my `client/src/style/AlbumDetail.css`, I have
```css
@import "~basscss/css/basscss.css";
@import '~react-soundplayer/styles/buttons.css';
@import '~react-soundplayer/styles/cover.css';
@import '~react-soundplayer/styles/icons.css';
@import '~react-soundplayer/styles/progress.css';
@import '~react-soundplayer/styles/volume.css';
...
/* other css classes */
...
```
Webpack crashes and returns 
```
ERROR in ./node_modules/css-loader??ref--1-1!./src/style/AlbumDetail.css
[1] Module not found: Error: Can't resolve '~basscss/css/basscss.css' in '/mnt/c/Users/ridhwaan/Source/homehost/client/src/styl
e'
[1]  @ ./node_modules/css-loader??ref--1-1!./src/style/AlbumDetail.css 3:10-96
[1]  @ ./src/style/AlbumDetail.css
[1]  @ ./src/components/AlbumDetail.js
[1]  @ ./src/components/Grid.js
[1]  @ ./src/pages/Music.js
[1]  @ ./src/App.js
[1]  @ ./src/index.js
[1]  @ multi ./node_modules/webpack-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/index.js
[1]
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Run `webpack` and `webpack-dev-server --inline --hot` to reproduce crash
<!-- A great way to do this is to provide your configuration via a GitHub gist. -->
<!-- Best provide a minimal reproduceable repo -->
<!-- If your issue is caused by a plugin or loader file the issue on the plugin/loader repo -->

**What is the expected behavior?**
Webpack should look at node_modules and resolve css

**If this is a feature request, what is motivation or use case for changing the behavior?**
css-loader fails to resolve node_module css.   
Is it failing because I have `exclude: /(node_modules|bower_components)/,` under `test: /\.js$/,` in `webpack.config.js`/?

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

my webpack.config.js

```javascript
const path = require('path');

var config = {
    entry: path.resolve(__dirname, 'src', 'index.js'),
    output: {
        path: path.resolve(__dirname, 'public'),
        filename: 'bundle.js'
    },
    devServer: {
        contentBase: path.resolve(__dirname, 'src'),
        publicPath: path.resolve(__dirname, 'public')
    },
    module: {
        rules: [
        {
            test: /\.js$/,
            exclude: /(node_modules|bower_components)/,
            use: [
                { loader: 'babel-loader',
                options: { presets: ['react','env'] } }
            ]
        },
        {
            test: /\.css$/,
            use: [
                {
                    loader: "style-loader", 
                    options: { singleton: true }
                },
                {
                    loader: "css-loader",
                    options: { modules: true }
                }
            ]
        }
        ]
    }
};
module.exports = config;
```


---

This issue was moved from webpack/webpack#6772 by @ooflorent. Original issue was by @ridhwaans.

--- a/lib/processCss.js
+++ b/lib/processCss.js
@@ -55,9 +55,11 @@ var parserPlugin = postcss.plugin("css-loader-parser", function(options) {
 				}
 				values.nodes[0].nodes.shift();
 				var mediaQuery = Tokenizer.stringifyValues(values);
-				if(loaderUtils.isUrlRequest(url, options.root) && options.mode === "global") {
+
+				if(loaderUtils.isUrlRequest(url, options.root)) {
 					url = loaderUtils.urlToRequest(url, options.root);
 				}
+
 				importItems.push({
 					url: url,
 					mediaQuery: mediaQuery

